src/components/Article.jsx

```diff
@@ -1,7 +1,42 @@
-import ArticleList from "./ArticleList";
+import "../index.css";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router";
+import { getArticleById } from "../api";
 
-function Article({ articles }) {
-  return <ArticleList articles={articles} />;
+function Article() {
+  const { articleId } = useParams();
+  const [article, setArticle] = useState(null);
+
+  useEffect(() => {
+    getArticleById(articleId)
+      .then((data) => {
+        setArticle(data);
+      })
+      .catch((err) => {
+        console.error("Error fetching article:", err);
```

Comment:
It would be nice if you had some sort of error state here rather than just throwing a console.error.